ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/ProceduralFilters.js

```diff
@@ -0,0 +1,470 @@
+    }
+    const siblings = Array.from(element.parentNode.children);
+    const otherHTMLElements = [];
+    for (const sib of siblings) {
+        if (sib === element) {
+            continue;
+        }
+        const siblingHTMLElement = _asHTMLElement(sib);
+        if (siblingHTMLElement !== null) {
+            otherHTMLElements.push(siblingHTMLElement);
+        }
+    }
+    return otherHTMLElements;
+};
+const _nextSiblingElement = (element) => {
+    if (!element.parentNode) {
+        return null;
+    }
+    const siblings = W.Array.from(element.parentNode.children);
+    const indexOfElm = siblings.indexOf(element);
+    const nextSibling = siblings[indexOfElm + 1];
+    if (nextSibling === undefined) {
+        return null;
+    }
+    return _asHTMLElement(nextSibling);
+};
+const _allChildren = (element) => {
+    return W.Array.from(element.children)
+        .map(e => _asHTMLElement(e))
+        .filter(e => e !== null);
+};
+const _allChildrenRecursive = (element) => {
+    return W.Array.from(element.querySelectorAll(':scope *'))
+        .map(e => _asHTMLElement(e))
+        .filter(e => e !== null);
+};
+const _stripCssOperator = (operator, selector) => {
+    if (selector[0] !== operator) {
+        throw new Error(`Expected to find ${operator} in initial position of "${selector}`);
+    }
+    return selector.replace(operator, '').trimStart();
+};
+// Implementation of ":css-selector" rule
+const operatorCssSelector = (selector, element) => {
+    const trimmedSelector = selector.trimStart();
+    if (trimmedSelector.startsWith('+')) {
+        const subOperator = _stripCssOperator('+', trimmedSelector);
+        if (subOperator === null) {
+            return [];
+        }
+        const nextSibNode = _nextSiblingElement(element);
+        if (nextSibNode === null) {
+            return [];
+        }
+        return nextSibNode.matches(subOperator) ? [nextSibNode] : [];
+    }
+    else if (trimmedSelector.startsWith('~')) {
+        const subOperator = _stripCssOperator('~', trimmedSelector);
+        if (subOperator === null) {
+            return [];
+        }
+        const allSiblingNodes = _allOtherSiblings(element);
+        return allSiblingNodes.filter(x => x.matches(subOperator));
+    }
+    else if (trimmedSelector.startsWith('>')) {
+        const subOperator = _stripCssOperator('>', trimmedSelector);
+        if (subOperator === null) {
+            return [];
+        }
+        const allChildNodes = _allChildren(element);
+        return allChildNodes.filter(x => x.matches(subOperator));
+    }
+    else if (selector.startsWith(' ')) {
+        return Array.from(element.querySelectorAll(':scope ' + trimmedSelector));
+    }
+    if (element.matches(selector)) {
+        return [element];
+    }
+    return [];
+};
+const _hasPlainSelectorCase = (selector, element) => {
+    return element.matches(selector) ? [element] : [];
+};
+const _hasProceduralSelectorCase = (selector, element) => {
+    var _a;
```

Comment:
SonarCloud flags this `var`.

It's there because this entire file is a compiled output from [brave-experiments/procedural-filters-js](https://github.com/brave-experiments/procedural-filters-js). We can update it once we have a way to use Typescript here and/or share the desktop implementation.